File: examples/monorepo/services/users/handler.js
// Users service — Node.js handlers

exports.list_users = async (event) => {
  const users = [
    { id: "u1", name: "Alice", email: "alice@example.com" },
    { id: "u2", name: "Bob", email: "bob@example.com" },
    { id: "u3", name: "Charlie", email: "charlie@example.com" },
  ];

  return {
    statusCode: 200,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      users,
      count: users.length,
      table: process.env.TABLE_NAME,
    }),
  };
};

exports.get_user = async (event) => {
  const userId = event.pathParameters?.userId || "unknown";
  return {
    statusCode: 200,
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      id: userId,
      name: "Alice",
      email: "alice@example.com",
      table: process.env.TABLE_NAME,
    }),
  };
};

File: examples/monorepo/services/products/handler.py
"""Products service — Python handlers"""

import json
import os


def list_products(event, context):
    products = [
        {"id": "p1", "name": "Widget", "price": 9.99},
        {"id": "p2", "name": "Gadget", "price": 24.99},
        {"id": "p3", "name": "Doohickey", "price": 14.99},
    ]
    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps({
            "products": products,
            "count": len(products),
            "table": os.environ.get("TABLE_NAME", "unknown"),
        }),
    }


def get_product(event, context):
    product_id = event.get("pathParameters", {}).get("productId", "unknown")
    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps({
            "id": product_id,
            "name": "Widget",
            "price": 9.99,
            "table": os.environ.get("TABLE_NAME", "unknown"),
        }),
    }
